### PR TITLE
[Bug 1130415] API for users with the most solutions in the last week.

### PR DESCRIPTION
**Pull request by mythmon**
This turned out to be more efficient than I expected, so there is no need to cache like I expected. I tried to do it all in one query. I'm pretty sure it would be possible in SQL, but I couldn't figure out how to get the ORM to do it. Oh well, this is pretty quick too.

r?

### CODE REVIEW

**Comment by rlr** [comment]
travis no likey.

File: kitsune/users/api.py [hunk 1]
```diff
@@ -346,6 +348,39 @@ def generate(self, request, **kwargs):
...
+            )
+
+        # Turn that list into a dictionary from username -> count.
+        username_to_count = {u['creator__username']: u['creator__count'] for u in raw_counts}
+
+        # Get all the profiles mentioned in the above.
+        profiles = Profile.objects.filter(user__username__in=username_to_count.keys())
+        result = ProfileFKSerializer(instance=profiles, many=True).data
+
+        # Pair up the profiles and the solutino counts.
```
Comment (rlr) [inline kitsune/users/api.py]: typo solutino

File: kitsune/users/api.py [hunk 2]
```diff
@@ -346,6 +348,39 @@ def generate(self, request, **kwargs):
...
+        # It looks like [{'creator__username': 'bob', 'creator__count': 12}, ...]
+        # This uses ``username`` instead of ``id``, because ``username`` appears
+        # in the output of ``ProfileFKSerializer``, whereas ``id`` does not.
+        raw_counts = (
+            Answer.objects
+            .exclude(solution_for=None)
+            .filter(created__gt=start)
+            .values('creator__username')
+            .annotate(Count('creator'))
+            )
```
Comment (rlr) [inline kitsune/users/api.py]: If this ends up being slow, you can use the contribution data. It should be very similar to this but filtering solutions only: https://github.com/mozilla/kitsune/blob/master/kitsune/community/utils.py#L17

**Comment by mythmon** [comment]
Fixed :point_up:

**Comment by rlr** [comment]
r+!

**Pull request merged**